hkmc2/shared/src/main/scala/hkmc2/semantics/Resolver.scala

```diff
@@ -354,9 +354,17 @@ class Resolver(tl: TraceLogger)
 
   def resolveDefn(defn: Definition)(using ICtx): ICtx =
   trace(s"Resolving definition: $defn"):
-    def traverseTermDef(tdf: TermDefinition) = tdf match
-    case TermDefinition(_, _, _, pss, tps, sign, body, _, TermDefFlags(isMethod), modulefulness, annotations) =>
-      def withCtxParams(pss: Ls[ParamList]): ICtx = pss
+    def traverseTermDef(tdf: TermDefinition) =
+      val TermDefinition(_, _, _, 
+        pss, tps, sign, body, 
+        _, TermDefFlags(isMethod), modulefulness, annotations
+      ) = tdf
```

Comment:
Ouch. Why not just access these fields as `tdf.X`? Doing this makes further changes to `TermDefinition` (such as adding or removing a field) more painful. Underscoreitis.

Reply:
I intended to write so because here we are traversing all subterms. If we make further changes to `TermDefinition`, like if we added some new subterms to `TermDefinition`, we would likely want to change here too!